apps/web-portal-client/lib/context/AppProvider.js

```diff
@@ -1,10 +1,17 @@
-import React from 'react'
+import React, {useEffect} from 'react'
+import {useLocation} from 'react-router-dom'
 import PropTypes from 'prop-types'
 import {client} from '../api'
 import {logger} from '../logger'
 import {AppContext} from './AppContext'
+import {BrowserKeys, BrowserStorage} from '../utils'
 
 export function AppProvider(props) {
+  const location = useLocation()
+  useEffect(() => {
```

Comment:
What about `AppRouter` component which would take  care about this side effect and redirecting to proper url when it exists

Reply:
if I correctly understand, I move this logic with the checkSession to one component

Reply:
I see, closing this one